runtime/src/main/java/org/corfudb/runtime/collections/ProtobufIndexer.java

```diff
@@ -249,7 +285,13 @@ private boolean isValidField(OneofDescriptor oneOfDescriptor, Message message, S
         }
 
         for (Message msg : messages) {
-            FieldDescriptor descriptor = msg.getDescriptorForType().findFieldByName(nestedIndexName);
+
+            if (!fdMapping.contains(idx)) {
+                throw new IllegalStateException("field " + nestedIndexName + " must be set!");
+            }
```

Comment:
Looks like this check can be factored-out of the loop.

Reply:
Good point. Will move it outside the loop.

Reply:
Done.